java_gen/pre-written/src/main/java/org/projectfloodlight/openflow/types/MacAddress.java

```diff
@@ -66,17 +72,14 @@ public static MacAddress of(long raw) {
      */
     @Nonnull
     public static MacAddress of(@Nonnull final String macString) throws IllegalArgumentException {
-        if (macString == null) {
-            throw new NullPointerException("macString must not be null");
-        }
+        Preconditions.checkArgument(!Strings.isNullOrEmpty(macString),
+                                    "macString must not be null or empty");
+        Preconditions.checkArgument(macString.length() == (STRING_LENGTH),
```

Comment:
The way you've rewritten the check here, you will be allocating the non-constant error string every time you check (See Gregor's email on Preconditions from a week or so ago).

I.e., you will *always* concatenate `FORMAT_ERROR + macString` to calculate the second parameter to `checkArgument`, even if the check does not fail. 

Either use the optional arguments as suggested by Gregor's email, or keep the original form with `if`.